src/libraries/System.Private.CoreLib/src/System/IO/FileStreamStrategy.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Win32.SafeHandles;
+
+namespace System.IO
+{
+    internal abstract class FileStreamStrategy : Stream
```

Comment:
Cool.  My hope with the `: Stream` part is that we'll be able to eschew the custom buffering done by FileStream and instead just instantiate a BufferedStream around the core FileStreamStrategy when buffering is needed.  I'm waving my hands a little, as BufferedStream isn't itself a FileStreamStrategy, but it could probably be wrapped in one, e.g. FileStream -> BufferedFileStreamStrategy -> BufferedStream -> SomeActualFileStreamStrategy.  Or at least consider this approach and whether it's feasible / would exhibit reasonable perf characteristics.